Hi @mgrbyte , I got the latest change on this branch, but test editing (create an uncloned gene and add sequence name and biotype) produces this error:

```
12:32:46.855 ERROR compojure.api.exception [qtp1579862569-65]
java.lang.ClassCastException: java.util.Arrays$ArrayList cannot be cast to java.util.Map$Entry
java.util.concurrent.ExecutionException: java.lang.ClassCastException: java.util.Arrays$ArrayList cannot be cast to java.util.Map$Entry
	at datomic.promise$throw_executionexception_if_throwable.invokeStatic(promise.clj:10)
	at datomic.promise$throw_executionexception_if_throwable.invoke(promise.clj:6)
	at datomic.promise$settable_future$reify__7074.deref(promise.clj:54)
	at clojure.core$deref.invokeStatic(core.clj:2312)
	at clojure.core$deref.invoke(core.clj:2298)
	at wormbase.names.gene$update_gene.invokeStatic(gene.clj:166)
	at wormbase.names.gene$update_gene.invoke(gene.clj:158)
	at wormbase.names.gene$fn__36068$fn__36073.invoke(gene.clj:422)
	at buddy.auth.accessrules$restrict$fn__24877.invoke(accessrules.clj:374)
	at compojure.api.resource$handle_sync.invokeStatic(resource.clj:96)
	at compojure.api.resource$handle_sync.invoke(resource.clj:91)
	at compojure.api.resource$create_handler$fn__23909.invoke(resource.clj:120)
	at compojure.api.routes.Route.invoke(routes.clj:90)
	at compojure.core$routing$fn__4768.invoke(core.clj:185)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:185)
	at compojure.core$routing.doInvoke(core.clj:182)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__4772.invoke(core.clj:192)
	at compojure.core$routing$fn__4768.invoke(core.clj:185)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:185)
	at compojure.core$routing.doInvoke(core.clj:182)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__4772.invoke(core.clj:192)
	at compojure.core$make_context$handler__4840.invoke(core.clj:285)
	at compojure.core$make_context$fn__4842.invoke(core.clj:293)
	at compojure.api.routes.Route.invoke(routes.clj:90)
	at compojure.core$routing$fn__4768.invoke(core.clj:185)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:185)
	at compojure.core$routing.doInvoke(core.clj:182)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__4772.invoke(core.clj:192)
	at compojure.api.routes.Route.invoke(routes.clj:90)
	at compojure.core$routing$fn__4768.invoke(core.clj:185)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:185)
	at compojure.core$routing.doInvoke(core.clj:182)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__4772.invoke(core.clj:192)
	at compojure.api.compojure_compat$make_context$handler__21433.invoke(compojure_compat.clj:26)
	at compojure.api.compojure_compat$make_context$fn__21435.invoke(compojure_compat.clj:34)
	at compojure.api.routes.Route.invoke(routes.clj:90)
	at compojure.core$routing$fn__4768.invoke(core.clj:185)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:185)
	at compojure.core$routing.doInvoke(core.clj:182)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__4772.invoke(core.clj:192)
	at compojure.api.compojure_compat$make_context$handler__21433.invoke(compojure_compat.clj:26)
	at compojure.api.routes.Route.invoke(routes.clj:90)
	at compojure.core$routing$fn__4768.invoke(core.clj:185)
	at clojure.core$some.invokeStatic(core.clj:2693)
	at clojure.core$some.invoke(core.clj:2684)
	at compojure.core$routing.invokeStatic(core.clj:185)
	at compojure.core$routing.doInvoke(core.clj:182)
	at clojure.lang.RestFn.applyTo(RestFn.java:139)
	at clojure.core$apply.invokeStatic(core.clj:659)
	at clojure.core$apply.invoke(core.clj:652)
	at compojure.core$routes$fn__4772.invoke(core.clj:192)
	at compojure.api.routes.Route.invoke(routes.clj:90)
	at ring.swagger.middleware$wrap_swagger_data$fn__22900.invoke(middleware.clj:35)
	at muuntaja.middleware$wrap_format$fn__18476.invoke(middleware.clj:73)
	at buddy.auth.middleware$wrap_authentication$fn__26894.invoke(middleware.clj:59)
	at buddy.auth.middleware$wrap_authorization$fn__26901.invoke(middleware.clj:110)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__17293.invoke(keyword_params.clj:36)
	at ring.middleware.params$wrap_params$fn__5000.invoke(params.clj:67)
	at ring.middleware.absolute_redirects$wrap_absolute_redirects$fn__30892.invoke(absolute_redirects.clj:47)
	at ring.middleware.content_type$wrap_content_type$fn__23460.invoke(content_type.clj:34)
	at ring.middleware.default_charset$wrap_default_charset$fn__30864.invoke(default_charset.clj:31)
	at ring.middleware.not_modified$wrap_not_modified$fn__23490.invoke(not_modified.clj:53)
	at wormbase.db$wrap_datomic$fn__24662.invoke(db.clj:80)
	at wormbase.names.service$wrap_not_found$fn__36224.invoke(service.clj:31)
	at ring.middleware.resource$wrap_resource$fn__30821.invoke(resource.clj:37)
	at ring.middleware.gzip$wrap_gzip$fn__36217.invoke(gzip.clj:94)
	at compojure.api.middleware$wrap_swagger_data$fn__19497.invoke(middleware.clj:190)
	at compojure.api.middleware$wrap_inject_data$fn__19475.invoke(middleware.clj:96)
	at muuntaja.middleware$wrap_params$fn__18472.invoke(middleware.clj:52)
	at compojure.api.middleware$wrap_exceptions$fn__19464.invoke(middleware.clj:62)
	at muuntaja.middleware$wrap_format_request$fn__18484.invoke(middleware.clj:114)
	at compojure.api.middleware$wrap_exceptions$fn__19464.invoke(middleware.clj:62)
	at muuntaja.middleware$wrap_format_response$fn__18488.invoke(middleware.clj:132)
	at muuntaja.middleware$wrap_format_negotiate$fn__18481.invoke(middleware.clj:96)
	at ring.middleware.keyword_params$wrap_keyword_params$fn__17293.invoke(keyword_params.clj:36)
	at ring.middleware.nested_params$wrap_nested_params$fn__17351.invoke(nested_params.clj:89)
	at ring.middleware.params$wrap_params$fn__5000.invoke(params.clj:67)
	at compojure.api.middleware$wrap_inject_data$fn__19475.invoke(middleware.clj:96)
	at compojure.api.routes.Route.invoke(routes.clj:90)
	at clojure.lang.Var.invoke(Var.java:381)
	at ring.middleware.reload$wrap_reload$fn__2752.invoke(reload.clj:39)
	at ring.middleware.stacktrace$wrap_stacktrace_log$fn__1209.invoke(stacktrace.clj:26)
	at ring.middleware.stacktrace$wrap_stacktrace_web$fn__1275.invoke(stacktrace.clj:96)
	at ring.adapter.jetty$proxy_handler$fn__485.invoke(jetty.clj:25)
	at ring.adapter.jetty.proxy$org.eclipse.jetty.server.handler.AbstractHandler$ff19274a.handle(Unknown Source)
	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
	at org.eclipse.jetty.server.Server.handle(Server.java:499)
	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:311)
	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:258)
	at org.eclipse.jetty.io.AbstractConnection$2.run(AbstractConnection.java:544)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:635)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$3.run(QueuedThreadPool.java:555)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.ClassCastException: java.util.Arrays$ArrayList cannot be cast to java.util.Map$Entry
	at user$eval36293$fn__36294.invoke(form-init7004740407507665327.clj:1)
	at datomic.error$deserialize_exception.invokeStatic(error.clj:135)
	at datomic.error$deserialize_exception.invoke(error.clj:117)
	at datomic.peer.Connection.notify_error(peer.clj:399)
	at datomic.connector$fn__9074.invokeStatic(connector.clj:169)
	at datomic.connector$fn__9074.invoke(connector.clj:167)
	at clojure.lang.MultiFn.invoke(MultiFn.java:233)
	at datomic.connector$create_hornet_notifier$fn__9080$fn__9081$fn__9084$fn__9085.invoke(connector.clj:194)
	at datomic.connector$create_hornet_notifier$fn__9080$fn__9081$fn__9084.invoke(connector.clj:189)
	at datomic.connector$create_hornet_notifier$fn__9080$fn__9081.invoke(connector.clj:187)
	at clojure.core$binding_conveyor_fn$fn__5476.invoke(core.clj:2022)
	at clojure.lang.AFn.call(AFn.java:18)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	... 1 common frames omitted
```

What do you think?

--- a/project.clj
+++ b/project.clj
@@ -27,7 +27,7 @@
     :aliases {"code-qa" ["do" ["eastwood"] "test"]
               "spec-test" ["run" "-m" "spec-test"]}
     :plugins
-    [[refactor-nrepl "2.4.0-SNAPSHOT"]
+    [[refactor-nrepl "2.4.0"]
      [cider/cider-nrepl "0.18.0"]
      [com.jakemccrary/lein-test-refresh "0.20.0"]
      [jonase/eastwood "0.2.4" :exclusions [org.clojure/clojure]]

--- a/src/wormbase/names/provenance.clj
+++ b/src/wormbase/names/provenance.clj
@@ -64,7 +64,7 @@
   [db tx-id]
   (d/pull db
           '[* {:provenance/what [:db/ident]
-               :provenance/who [:person/id :person/name]
+               :provenance/who [:person/email :person/name :person/id]
                :provenance/how [:db/ident]
                :provenance/split-from [:gene/id]
                :provenance/split-into [:gene/id]
@@ -77,15 +77,14 @@
   [data]
   (w/postwalk (fn presenter [m]
                 (cond
-                  (and (map? m) (:db/ident m))
-                  (:db/ident m)
+                  (and (map? m) (:db/ident m)) (:db/ident m)
                   (map? m) (dissoc m :db/id :db/txInstant)
                   :default m))
               data))
 
 (defn query-provenance
-  "query for the entire history of an entity `entity-id`.
-  Note that lookup-ref can be used instead of a numeric identifier."
+  "Query for the entire history of an entity `entity-id`.
+  `entity-id` can be a datomic id number or a lookup-ref."
   [db entity-id]
   (let [pull (partial pull-provenance db)
         sort-mrf #(sort-events-by-when % :most-recent-first true)]

